pennylane/templates/subroutines/trotter.py

```diff
@@ -494,6 +505,77 @@ def compute_decomposition(*args, **kwargs):
         return decomp
 
 
+def _trotter_product_decomposition_resources(n, order, ops):
+    def _recursive(order, ops):
+        reps = Counter()
+        if order == 1:
+            for op in ops:
+                reps[resource_rep(qml.ops.op_math.Exp, base=op, num_steps=None)] += 1
+            return reps
+
+        if order == 2:
+            for op in ops + ops[::-1]:
+                reps[resource_rep(qml.ops.op_math.Exp, base=op, num_steps=None)] += 1
+            return reps
+
+        ops_ctr_1 = _recursive(order - 2, ops)
+        ops_ctr_2 = _recursive(order - 2, ops)
+
+        for key in ops_ctr_1:
+            ops_ctr_1[key] = ops_ctr_1[key] * 4
+
+        ops_ctr_1.update(ops_ctr_2)
+
+        return ops_ctr_1
+
+    resources = _recursive(order, ops)
+
+    for _ in range(n - 1):
+        for key in resources:
+            resources[key] += 1
+
+    return resources
```

Comment:
Instead of replicating the decomposition logic, would you be able to actually find a relationship between the number of each type of op with `n` and `order`? For example, see
https://github.com/PennyLaneAI/pennylane/blob/5c7e16bc85df482eaa718dbcb9e972e021f27005/pennylane/ops/op_math/decompositions/controlled_decompositions.py#L467
It may be possible to calculate the number of each type of operator from `n` and `order` using an equation rather than going through with the entire decomposition and counting things out.